studio/components/LanguageSelector.tsx

```diff
@@ -0,0 +1,121 @@
+import { Box, Button, Card, Checkbox, Flex, Text, useTheme } from "@sanity/ui";
+import React, { useMemo } from "react";
+import { PatchEvent, set } from "sanity";
+import {
+  Language,
+  supportedLanguages,
+} from "internationalization/supportedLanguages";
+
+interface LanguageSelectorProps {
+  value?: Language[];
+  onChange: (event: PatchEvent) => void;
+}
+
+const colorMap = {
+  dark: {
+    selected: "#29356a",
+    default: "rgba(19, 20, 27, 0.5)",
+  },
+  light: {
+    selected: "#E3F2FD",
+    default: "white",
+  },
+};
+
+const LanguageSelector = ({ value = [], onChange }: LanguageSelectorProps) => {
+  const theme = useTheme();
+  const prefersDark = theme.sanity.v2?.color._dark ?? false;
+  const themeType = prefersDark ? "dark" : "light";
+
+  // Find current default language
+  const currentDefaultLanguage = value.find((lang) => lang.default)?.id || null;
+
+  const handleLanguageSelection = (lang: Language) => {
+    const isSelected = value.some((item) => item.id === lang.id);
+    let updatedValue;
+
+    if (isSelected) {
+      // Deselect language
+      updatedValue = value.filter((item) => item.id !== lang.id);
+
+      // If deselecting the default language, assign a new default
+      if (currentDefaultLanguage === lang.id && updatedValue.length > 0) {
+        updatedValue[0] = { ...updatedValue[0], default: true }; // Set the first language as default
+      }
+    } else {
+      // Add new language, without making it the default
+      updatedValue = [...value, { ...lang, default: false }];
+    }
+
+    onChange(PatchEvent.from(set(updatedValue)));
+  };
+
+  const handleDefaultSetting = (lang: Language) => {
+    const updatedValue = value.map((item) => ({
+      ...item,
+      default: item.id === lang.id, // Only one language can be the default
+    }));
+    onChange(PatchEvent.from(set(updatedValue)));
+  };
+
+  // Memoize value for performance and reusability
+  const languageCards = useMemo(
```

Comment:
somewhat uncertain if useMemo is really needed. bob bob 🤷‍♀️

Reply:
There are really no expensive operations here, so I think we should drop it for now.